modules/navier_stokes/src/kernels/INSBase.C

```diff
@@ -326,3 +328,69 @@ INSBase::dTauDUComp(unsigned comp)
          2. * (2. * U.norm() / h) * 2. / h * U(comp) * _phi[_j][_qp] /
          (U.norm() + std::numeric_limits<double>::epsilon());
 }
+
+RealVectorValue
+INSBase::strongViscousTermLaplaceRZ() const
+{
+  // To understand the code below, visit
+  // https://en.wikipedia.org/wiki/Del_in_cylindrical_and_spherical_coordinates.
+  // The u_r / r^2 term comes from the vector Laplacian. The -du_r/dr * 1/r term comes from
+  // the scalar Laplacian. The scalar Laplacian in axisymmetric cylindrical coordinates is
+  // equivalent to the Cartesian Laplacian plus a 1/r * df/dr term. And of course we are
+  // applying a minus sign here because the strong form is -\nabala^2 * \vec{u}
+
+  const auto r = _q_point[_qp](_rz_radial_coord);
+  RealVectorValue rz_term;
+  rz_term(0) = -_mu[_qp] * _grad_u_vel[_qp](_rz_radial_coord) / r;
+  rz_term(1) = -_mu[_qp] * _grad_v_vel[_qp](_rz_radial_coord) / r;
+  mooseAssert((_rz_radial_coord == 0) || (_rz_radial_coord == 1),
+              "We expect X or Y as the possible radial coordinate");
+  if (_rz_radial_coord == 0)
+    rz_term(0) += _mu[_qp] * _u_vel[_qp] / (r * r);
+  else
+    rz_term(1) += _mu[_qp] * _v_vel[_qp] / (r * r);
```

Comment:
can one line this?
with 
rz_term(_rz_radial_coord)

Reply:
It uses different variables

Reply:
ah good point.